cmd/eksctl-anywhere/cmd/validatecreatecluster.go

```diff
@@ -0,0 +1,140 @@
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/dependencies"
+	"github.com/aws/eks-anywhere/pkg/executables"
+	"github.com/aws/eks-anywhere/pkg/kubeconfig"
+	"github.com/aws/eks-anywhere/pkg/types"
+	"github.com/aws/eks-anywhere/pkg/validations"
+	"github.com/aws/eks-anywhere/pkg/validations/commandvalidations"
+	"github.com/aws/eks-anywhere/pkg/validations/createvalidations"
+	"github.com/aws/eks-anywhere/pkg/version"
+)
+
+type validateOptions struct {
+	clusterOptions
+	hardwareCSVPath       string
+	tinkerbellBootstrapIP string
+}
+
+var valOpt = &validateOptions{}
+
+var validateCreateCmd = &cobra.Command{
+	Use:          "create -f <cluster-config-file> [flags]",
+	Short:        "Validate create cluster",
+	Long:         "This command is used to validate the create cluster action",
+	PreRunE:      bindFlagsToViper,
+	SilenceUsage: true,
+	RunE:         valOpt.validateCreateCluster,
+}
+
+func init() {
+	validateCmd.AddCommand(validateCreateCmd)
+	validateCreateCmd.Flags().StringVarP(&valOpt.fileName, "filename", "f", "", "Filename that contains EKS-A cluster configuration")
+	validateCreateCmd.Flags().StringVarP(
+		&valOpt.hardwareCSVPath,
+		TinkerbellHardwareCSVFlagName,
+		TinkerbellHardwareCSVFlagAlias,
+		"",
+		TinkerbellHardwareCSVFlagDescription,
+	)
+	validateCmd.Flags().StringVar(&valOpt.tinkerbellBootstrapIP, "tinkerbell-bootstrap-ip", "", "Override the local tinkerbell IP in the bootstrap cluster")
+
+	if err := validateCreateCmd.MarkFlagRequired("filename"); err != nil {
+		log.Fatalf("Error marking flag as required: %v", err)
+	}
+}
+
+func (valOpt *validateOptions) validateCreateCluster(cmd *cobra.Command, _ []string) error {
+	ctx := cmd.Context()
+
+	clusterSpec, err := cluster.NewSpecFromClusterConfig(valOpt.fileName, version.Get())
+	if err != nil {
+		return err
+	}
+
+	if clusterSpec.Config.Cluster.Spec.DatacenterRef.Kind == v1alpha1.TinkerbellDatacenterKind {
```

Comment:
if this is copy paste from `create` I would vote to at least put it in a method

Reply:
Will do, I'll group this in with the cmd flags PR